gnucash/gnome/gnc-budget-view.c

```diff
@@ -1380,20 +1397,17 @@ gnc_budget_view_refresh(GncBudgetView *view)
     /* Create any needed columns */
     while (num_periods_visible < num_periods)
     {
-        GtkCellRenderer* renderer;
-
-        col = gnc_tree_view_account_add_custom_column(
+        GtkCellRenderer *renderer = gnc_cell_renderer_text_flag_new ();
+        g_object_set(renderer, "flag-color-rgba", &flag_color, NULL);
```

Comment:
I suggest to keep all renderer configuration together in the call to budget_col_source function. This will make even more sense if you reuse the selection and disabled background colors.

Reply:
I could use some help here, as I'm not much of an expert on gtk.
I can't find a simple way to get the treview's current "selected" color and the current row's color.
Some hints?

Reply:
You can pretty much reuse the code you have in your `get_flag_color` function. However in this case you get the style context from the budgetview widget.
The colors you're looking for are background colors which you can retrieve with `gtk_style_context_get_background_color`

The background color used for highlighting rows is the one for the `GTK_STATE_FLAG_SELECTED` state. The background color for other rows is the one you get for the `GTK_STATE_FLAG_NORMAL` state.

I think you can forget what I wrote about the dimmed state. That probably doesn't apply here as we never disable the treeview widget.

Reply:
Ok, understood.

But in gtk doc it states:
```
gtk_style_context_get_background_color has been deprecated since version 3.16 and should not be used in newly-written code.
Use gtk_render_background() instead.
```

That’s why I originally discarded this function, but the doc proposed `gtk_render_background` doesn’t apply to our use case.

Should we proceed with a function marked as deprecated?

Reply:
Probably not. However there's also the more generic and non-deprecated gtk_style_context_get, which you can use to achieve the same thing.

Reply:
done